File: 2134. Minimum Swaps to Group All 1's Together II/Solution.py
class Solution:
    def minSwaps(self, nums: List[int]) -> int:
        windowSize = sum(nums)
        ones = 0
        for i in range(windowSize):
            ones += nums[i]
        swaps = windowSize - ones
        for i in range(len(nums)):
            ones += nums[(i+windowSize)%len(nums)]
            ones -= nums[i]
            swaps = min(swaps, windowSize - ones)
        return swaps

File: 2134. Minimum Swaps to Group All 1's Together II/Solution.java
class Solution {
    public int minSwaps(int[] nums) {
        int windowSize = 0;
        for(int i=0;i<nums.length;i++){
            windowSize += nums[i];
        }
        int ones = 0;
        for(int i=0;i<windowSize;i++){
            ones += nums[i];
        }
        int swaps = windowSize - ones;
        for(int i=0;i<nums.length;i++){
            ones -= nums[i];
            ones += nums[(i+windowSize)%nums.length];
            swaps = Math.min(swaps, windowSize - ones);
        }
        return swaps;
    }
}
